### PR TITLE
Picture element

### LINKED ISSUE #55: any plans to support the <picture> element?
is supporting `<picture>` in the roadmap at all?

i've hacked this together locally using [processImage](https://github.com/imgix/react-imgix/blob/master/src/support.js) but i'm guessing that's not meant for public consumption. i could try to put a pr together but i'm still sorting through what the api would look like..

### PR DESCRIPTION
**Pull request by modosc**
first pass, still testing it out in our app. 

Closes #55

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,11 +1,15 @@
 import ReactDOM from 'react-dom'
 import React, {Component, PropTypes} from 'react'
+import { deprecate } from 'react-is-deprecated'
```
Comment (frederickfogerty) [inline src/index.js]: Awesome! I was just thinking we should do this, and you've already done it!

File: README.md [hunk 2]
```diff
@@ -19,25 +19,61 @@ npm install --save react-imgix
...
+### height={number}
+force images to be a certain height, overrides `precision`
+### width={number}
+force images to be a certain width, overrides `precision`
+### generateSrcSet={bool} 
+generate `2x` and `3x` src sets when using an `<img>` tag. Defaults to `true`
+### customParams={object}
+any other Imgix params to add to the image `src`
+### imgProps={object}
+any other attributes to add to the html node (example: `alt`, `data-*`, `className`)
```
Comment (frederickfogerty) [inline README.md]: These docs are great!

File: src/index.js [hunk 3]
```diff
@@ -128,28 +139,121 @@ export default class ReactImgix extends Component {
...
 
-      childProps.src = _src
+          _children.push(React.createElement(this.constructor, imgProps))
+        } else if (imgIdx !== (_children.length - 1)) {
+          // found one, need to move it to the end
+          _children.splice(_children.length - 1, 0, _children.splice(imgIdx, 1)[0])
+        }
+        break
+      default:
+        // TODO - should we console.warn something here?
```
Comment (frederickfogerty) [inline src/index.js]: I think `PropTypes` should handle the warning here - if they're ignoring that, they'd probably ignore our warning too.

File: src/index.js [hunk 4]
```diff
@@ -128,28 +139,121 @@ export default class ReactImgix extends Component {
...
+          delete imgProps.styles
+
+          // have to strip out props set to undefined since they will override
+          // any defaultProps in the child
+          Object.keys(imgProps).forEach(k => {
+            if (imgProps[k] === undefined) delete imgProps[k]
+          })
 
-      childProps.src = _src
+          _children.push(React.createElement(this.constructor, imgProps))
```
Comment (frederickfogerty) [inline src/index.js]: Is `React.createElement` analogous to using JSX? What's the difference?

File: src/index.test.js [hunk 5]
```diff
@@ -46,8 +48,196 @@ describe('<img> mode - pre-mount', () => {
...
+      expect(children.length).toBe(1)
+    })
+    it('should not pass props down to children', () => {
+      expect(lastChild.props.faces).toBe(undefined)
+      expect(lastChild.props.entropy).toBe(undefined)
+    })
+  })
+})
+
+const shouldBehaveLikeBg = function () {
```
Comment (frederickfogerty) [inline src/index.test.js]: 👍

**Comment by frederickfogerty** [comment]
Looks like Travis is having a problem with `.findIndex`. Maybe we need to add come config to babel, or add a [polyfill](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex)

**Comment by modosc** [comment]
polyfill added.

File: src/index.js [hunk 6]
```diff
@@ -1,3 +1,5 @@
+import polyfill from './array-findindex' // eslint-disable-line no-unused-vars
```
Comment (frederickfogerty) [inline src/index.js]: This can just be `import './array-findindex'`

**Comment by modosc** [comment]
so far so good - i think the naming of`imgProps` is still confusing but beyond that things are working like i'd expect.

**Comment by modosc** [comment]
ping?

**Comment by frederickfogerty** [comment]
Hey @modosc. Are you happy with this? I'll have another look if you're done with this.

**Comment by modosc** [comment]
yeah, we're using it in production now and it's simplified a lot of our code. seems ok so far?

**Comment by frederickfogerty** [comment]
Looks good to me! I've just updated the README a bit to make it a bit cleaner, and I've put some changes of my own in there too. Thanks so much for this PR @modosc!

**Comment by frederickfogerty** [comment]
@modosc publish to npm as version `5.2.0` 🎉

**Pull request merged**